templates/index.html

```diff
@@ -2,254 +2,709 @@
+  */
+  QueryBuilder.LinkGen.getSampleSize = function(){
+    var sampleSize = $('#qb-sample-size-list input:checked').val();
+
+    if(sampleSize === '50 rows'){
+      return '';
+    } else {
+      return 'stream=True'
+    }
+  };
+
+  /*
+    Obtain the calendar date
+  */
+  QueryBuilder.LinkGen.getCalendarDate = function(id, name){
+    var calendarDate = $('#' + id).val();
+
+    if(calendarDate.trim() === ''){
+      return '';
+    } else {
+      return name + '=' + calendarDate;
+    }
+  };
+
+  /*
+    Obtain info for a multi-select.
+  */
+  QueryBuilder.LinkGen.getMultiSelectInfo = function(id, name){
+    var separator = '%7C';
+    var multiSelect = $('[name*="' + id + '"]');
+    var selectedOptions = $('[name*="' + id + '"] option:checked');
+    var values = [];
+
+    // have to use forEach because querySelectorAll is a NodeList which doesn't have a `map()` function
+    selectedOptions.each(function(){
+        var value = $(this).val();
+
+      // do not include the 'None' values
+      if(value !== ''){
+        values.push(value);
+      }
+    });
+
+    if(values.length === 0){
+      return '';
+    }
+
+    return name + '=' + values.join(separator);
+  };
+
+  /*
+    Convert an array of query string sections into a query string.
+  */
+  QueryBuilder.LinkGen.createQueryString = function(sections){
+    var filteredSections = sections.filter(function(section){
+      return section !== '';
+    });
+
+    if(filteredSections.length === 0){
+      return '';
+    }
+
+    return '?' + filteredSections.join('&');
+  };
+
+  /*
+    Generate the link from the page.
+  */
+  QueryBuilder.LinkGen.createLink = function(fileType){
+    var baseURL = 'http://datastore.iatistandard.org/api/1/access/';
+    var format = QueryBuilder.LinkGen.getFormat();
+    var grouping = QueryBuilder.LinkGen.getGrouping();
+    var reportingOrgs = QueryBuilder.LinkGen.getMultiSelectInfo('1922375458', 'reporting-org');
+    var reportingOrgTypes = QueryBuilder.LinkGen.getMultiSelectInfo('18398991', 'reporting-org.type');
+    var sectors = QueryBuilder.LinkGen.getMultiSelectInfo('1954968791', 'sector');
+    var recipientCountries = QueryBuilder.LinkGen.getMultiSelectInfo('605980212', 'recipient-country');
+    var recipientRegions = QueryBuilder.LinkGen.getMultiSelectInfo('1179181326', 'recipient-region');
+    var start_date__lt = QueryBuilder.LinkGen.getCalendarDate('start_date__lt', 'start-date__lt');
+    var start_date__gt = QueryBuilder.LinkGen.getCalendarDate('start_date__gt', 'start-date__gt');
+    var end_date__lt = QueryBuilder.LinkGen.getCalendarDate('end_date__lt', 'end-date__lt');
+    var end_date__gt = QueryBuilder.LinkGen.getCalendarDate('end_date__gt', 'end-date__gt');
+    var transactionProviderOrg = QueryBuilder.LinkGen.getMultiSelectInfo('transaction_provider_org', 'transaction_provider-org');
+    var participating_org = QueryBuilder.LinkGen.getMultiSelectInfo('participating_org', 'participating_org');
```

Comment:
should be `participating-org`

Reply:
why?

Reply:
[see here](http://datastore.iatistandard.org/docs/api/#participating-org)

Reply:
Good spot!